src/InverseKinematics.cpp

```diff
@@ -0,0 +1,154 @@
+#include "InverseKinematics.h"
+
+
+InverseKinematics::InverseKinematics()
+:updated(false)
+,initialized(false)
+,alreadyOptimized(false)
+{
+    solverPointer = new InverseKinematicsIPOPT();
+    loader = IpoptApplicationFactory();
```

Comment:
The point is that an object of ``InverseKinematicsIPOPT`` should be passed to an object of ``IpoptApplicationFactory``, and destroyed after compuations has been made, so how can I do that from the same class? I designed those two classes as a couple for that reason, and unfortunately cannot be easily separable I think.